src/frontend/apps/deposit/components/Dashboard/common/DepositedFileRow/index.tsx

```diff
@@ -0,0 +1,181 @@
+import { Anchor, Box, Grid, ResponsiveContext, Text } from 'grommet';
+import React, { useContext } from 'react';
+import { defineMessages, useIntl } from 'react-intl';
+import { DateTime } from 'luxon';
+
+import { useUpdateDepositedFile } from 'apps/deposit/data/queries';
+import { DepositedFile } from 'apps/deposit/types/models';
+import { bytesToSize } from 'apps/deposit/utils/bytesToSize';
+import { truncateFilename } from 'apps/deposit/utils/truncateFilename';
+
+const messages = defineMessages({
+  labelDownload: {
+    defaultMessage: 'Download',
+    description: 'Button label for downloading a deposited file.',
+    id: 'component.DashboardFileDepositoryStudent.labelDownload',
+  },
+});
+
+interface DepositedFileProps {
+  file: DepositedFile;
+}
+
+export const DepositedFileRow = ({ file }: DepositedFileProps) => {
+  const intl = useIntl();
+  const size = useContext(ResponsiveContext);
+  const uploadedOn = file.uploaded_on
+    ? DateTime.fromISO(file.uploaded_on)
+    : null;
+  const uploadedOnDate = uploadedOn ? uploadedOn.toFormat('dd/MM/yyyy') : null;
+  const uploadedOnTime = uploadedOn ? uploadedOn.toFormat('HH:mm') : null;
+
+  const { mutate } = useUpdateDepositedFile(file.id);
```

Comment:
The onSuccess should replace the current file ? Otherwise I think the component never know that the `read` property has changed and there is no re-render and no change in hte UI showing that the file is read.

Reply:
Sorry I don't get what you mean.
I don't see any problem by testing this.

Reply:
When I download the file, the file row doesn't change to it's `read` state and continue to be white.

Reply:
Oh, ok, I see